.github/workflows/reusable_promote_prerelease.yaml

```diff
@@ -0,0 +1,56 @@
+name: Promote Prerelease to Release
+
+# This workflow checks for the latest successful prerelease, excluding any labeled with "artifacts-pending" or "pre-release-failure".
+# If a suitable prerelease is found, it promotes it to a full release. This action should consequently trigger any workflows
+# that respond to the 'released' event type, ultimately invoking 'reusable_on_release.yaml'. This facilitates an automated
+# continuation of the release process.
+
+#  Usage from the caller workflow:
+#  jobs:
+#    promote_prerelease:
+#      uses: newrelic/coreint-automation/.github/workflows/reusable_promote_prerelease.yaml@v2
+#      secrets:
+#        bot_token: ${{ secrets."github_token_name" }}
+
+on:
+  workflow_call:
+    inputs:
+      block_endpoint:
+        description: This endpoint should respond status 200 for the promotion to continue.
+        required: false
+        type: string
+        default: https://newrelic.github.io/coreint-automation/automatic_release_enable
+    secrets:
+      bot_token:
+        description: "github token"
+        required: true
+
+jobs:
+  check_block_endpoint:
+    name: Check block endpoint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check block endpoint
+        run: |
+          http_status=$(curl -s -o /dev/null -w "%{http_code}" "${{ inputs.block_endpoint }}")
+
+          if [ "$http_status" != 200 ]; then
+            echo "Fail checking release block endpoint: ${{ inputs.block_endpoint }}"
+            exit 1
+          fi
+
+  promote_prerelease:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Promote previous prerelease to release
+        env:
+          GH_TOKEN: "${{ secrets.bot_token }}"
+        run: |
+          PRERELEASE_INFO=$(gh release list --repo ${{ github.repository }} --limit 1 --prerelease | grep -v "artifacts-pending" | grep -v "pre-release-failure")
+          if [ ! -z "$PRERELEASE_INFO" ]; then
+            PRERELEASE_TAG=$(echo $PRERELEASE_INFO | cut -d$'\t' -f3)
+            echo "Prerelease $PRERELEASE_TAG is ready for promotion."
+            gh release edit "$PRERELEASE_TAG" --repo ${{ github.repository }} --draft=false --prerelease=false
+          else
+            echo "No prerelease ready for promotion."
+          fi
```

Comment:
have we decide not to wait that the release is X days old before upgrading?

Reply:
Since this is run weekly, the pre-release will be available for at least 1 week, won't it?

First, we check if there's any pending prerelease to be promoted to release, if any, we promote it. Then, we try to create a prerelease that is going to be available for 1 week.

So, the only scenario where both could trigger is if there's a pending prerelease to be promoted and a new prerelease to be created, but I think that's ok. However, for any prerelease, we'll always wait until the next scheduled time.

Reply:
The idea of this reusable workflow is to be used like this https://github.com/newrelic/nri-haproxy/pull/135 so we control the releasing process in the same cron as the prerelease.

First, we check if there's any pending prerelease to be promoted to release, if any, we promote it. Then, we try to create a prerelease that is going to be available for 1 week.

So, the only scenario where both could trigger is if there's a pending prerelease to be promoted and a new prerelease to be created, but I think that's ok. However, for any prerelease, we'll always wait until the next scheduled time. LMKWYT.